rfcs/0001-connection-pooling.md

```diff
@@ -0,0 +1,71 @@
+---
+feature: Connection Pooling
+start-date: 2021-02-04
+author: kiwicopple
+co-authors: steve-chavez, dragarcia
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+We would like to explore connection pooling on Supabase. This RFC is intended to decide:
+
+- Whether we should provide a pooler
+- Which connection pooler we should use
+- Where in the stack it would be installed - i.e. if should bundle it with the Postgres build
+
+
+# Motivation
+[motivation]: #motivation
+
+In Postgres, every connection is a process. Because of this, a lot of connections to the database can be very expensive on memory. 
+
+When connecting to Postgres database from serverless functions, there is no connection pooling, and so the server needs to maintain hundreds/thousands of connections.
+
+
+# Detailed design
+[design]: #detailed-design
+
+This is still in the "Gather Feedback" stage. To start the discussion:
+
+
+### 1. Decide on a PG Pooler
+
+- `pg_bouncer` - https://www.pgbouncer.org/
```

Comment:
I'd vote for pgbouncer. Heroku(1) and DO(2) use it.

(1):https://www.digitalocean.com/community/tutorials/managed-databases-connection-pools-and-postgresql-benchmarking-using-pgbench (search for pgbouncer)
(2):https://devcenter.heroku.com/articles/best-practices-pgbouncer-configuration

Reply:
There's also an AWS pgbouncer fork: https://github.com/awslabs/pgbouncer-rr-patch.